go-controller/pkg/node/management-port-dpu.go

```diff
@@ -14,49 +14,61 @@ import (
 	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
 )
 
-type managementPortDPU struct {
+type managementPortRepresentor struct {
 	nodeName    string
 	hostSubnets []*net.IPNet
-	vfRepName   string
+	repName     string
 }
 
-// newManagementPortDPU creates a new managementPortDPU
-func newManagementPortDPU(nodeName string, hostSubnets []*net.IPNet) ManagementPort {
-	return &managementPortDPU{
+// newManagementPortRepresentor creates a new managementPortRepresentor
+func newManagementPortRepresentor(nodeName string, hostSubnets []*net.IPNet) ManagementPort {
+	var repName string
+
+	// In ovnkube-node mode DPU representor name stored in MgmtPortNetdev variable
+	if config.OvnKubeNode.MgmtPortRepresentor == "" {
```

Comment:
It sounds confusing to store netdev name in MgmtPortRepresentor struct, would it be possible to avoid?

Reply:
Not sure if I understood correctly. It is original design. In DPU mode netdevice name of representor passed as mgmt-port-netdev variable. In DPU host mode, same variable used to pass netdevice name of the function. In both cases netdevice name stored in corresponding management port structure.

Reply:
I guess when the `ovnkube-node-mgmt-port-representor` flag is introduced, it changed the meaning of `ovnkube-node-mgmt-port-netdev` to refer only to VF netdevice, instead of both netdevice and rep.

Reply:
Maybe, but I kept it as is for backward compatibility.